src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs

```diff
@@ -683,56 +684,56 @@ private int ExecuteEvmCall<TTracingInst>(
 
             ExecutionType executionType = tx.IsContractCreation ? ((spec.IsEofEnabled && tx.IsEofContractCreation) ? ExecutionType.TXCREATE : ExecutionType.CREATE) : ExecutionType.TRANSACTION;
 
-            using (EvmState state = EvmState.RentTopLevel(gasAvailable, executionType, in env, in accessedItems, in snapshot))
+            using (EvmState state = EvmState.RentTopLevel(gasAvailable, executionType, env, in accessedItems, in snapshot))
```

Comment:
It may be worth adding `env.Return()` if `EvmState` was not created (as of now only its `Dispose` call is doing `Return` here).

Probably always under `FailContractCreate` label, and maybe under some condition under `Complete` label.